yt_dlp/extractor/rheinmaintv.py

```diff
@@ -0,0 +1,601 @@
+                    'duration': 3475200000,
+                    'timescale': 10000000,
+                    'width': 1920,
+                    'height': 1080,
+                    'fourcc': 'H264',
+                    'language': 'und',
+                    'codec_private_data': '0000000167640028ACD940780227E5C044000003000400000300C83C60C6580000000168EBECB22C',
+                    'sampling_rate': None,
+                    'channels': 2,
+                    'bits_per_sample': 16,
+                    'nal_unit_length_field': 4
+                }
+            }],
+            'subtitles': {},
+            'thumbnail': 'https://rmtvmedia0.blob.core.windows.net/e1cf7655-91b9-4b7e-b19f-a79852c7b41f/06_Standard.jpg?sv=2016-05-31&sr=c&sig=jQPJSdxCxJMX8yUZkFUPNrv1qISb0pCouK9MzAMSOMc%3D&st=2022-11-14T14%3A50%3A02Z&se=3022-11-15T14%3A50%3A02Z&sp=r',
+            'timestamp': 1668527402,
+            'duration': 348.0,
+            'view_count': int,
+            'upload_date': '20221115'
+        }
+    }]
+
+    def _real_extract(self, url):
+        mobj = self._match_valid_url(url)
+        display_id = mobj.group('display_id')
+        video_id = 'vom %s' % mobj.group('date')
+        if mobj.group('serial_number'):
+            video_id += ' (%s)' % mobj.group('serial_number')
+        webpage = self._download_webpage(url, video_id)
+
+        headline = self._html_search_regex(r'<h1><span class="title">([^<]*)</span>', webpage, 'headline')
+
+        source, img = self._search_regex(r'(?s)(?P<source><source[^>]*>)(?P<img><img[^>]*>)', webpage, 'video', group=('source', 'img'))
+        source = extract_attributes(source)
+        img = extract_attributes(img)
+
+        # json = self._search_json_ld(webpage, video_id)  # uncomment to extract the useless 'contentUrl' (as 'url') instead of the essential 'embedUrl'
+        #                                                 # then use get(json, 'title') instead of get(json, 'name')
+        json = self._parse_json(self._search_regex(r'(?s)<script type="application/ld\+json">([^<]*)</script>', webpage, 'json'), video_id)
+        # json = None  # uncomment to go without ld+json altogether
+
+        def get(dictionary, *keys):  # return ..utils.traverse_obj(dictionary, keys)
+            for k in keys:
+                if dictionary is None:
+                    return None
+                dictionary = dictionary.get(k)
+            return dictionary
+
+        ism_manifest_url = source['src'] or get(json, 'embedUrl')  # attribute 'src' is mandatory
+        formats, subtitles = self._extract_ism_formats_and_subtitles(ism_manifest_url, video_id)
+
+        def removeprefix(string, prefix):
+            return string[len(prefix):] if string.startswith(prefix) else None
+
+        def extract_format(internet_media_type, media):  # subtype of media (aka type)
+            return removeprefix(internet_media_type, media + '/') if internet_media_type is not None else None
```

Comment:
pointless functions. See utils

Reply:
Any suggestions?

Reply:
Use `traverse_obj` instead. See the [contributing.md](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#mandatory-and-optional-metafields) for further info. For example `get(d, 'a', 'b')` can be written as `traverse_obj(d, ('a', 'b'))`.

Reply:
traverse_obj is of no use here.

Reply:
I am showing you replacements for future reference (`get` => `utils.traverse_obj`, `removeprefix` => `utils.remove_start`, `extract_format` => `utils.mimetype2ext`).

`self._extract_ism_formats_and_subtitles(ism_manifest_url, video_id)` should already handle setting `ext`, do you need to overwrite these? If so, that would be worth a comment. In that case you should use `mimetype2ext`.

Reply:
OMG, I overlooked mimetype2ext despite its pretty straightforward name. Thank you!

This removes one concern about this part of the code. The other is, as you mention, that it blindly overwrites the default extension. Apparently, .ismv (for video, .isma for audio) is hardwired in self._extract_ism_formats_and_subtitles. Of course, the file could be converted afterwards or even by yt-dlp itself when asked to do so. But I'd prefer to give it the original (?) extension right away. It always seems to be .mp4 for this site.